ansible-collection/plugins/modules/px_backup/backup_schedule.py

```diff
@@ -0,0 +1,494 @@
+        params['enumerate_options.name_filter']= enumerate_options.get('enumerate_options.name_filter')
+        params['enumerate_options.cluster_name_filter']= enumerate_options.get('enumerate_options.cluster_name_filter')
+        params['enumerate_options.object_index']= enumerate_options.get('enumerate_options.object_index')
+        params['enumerate_options.include_detailed_resources']= enumerate_options.get('enumerate_options.include_detailed_resources')
+        params['enumerate_options.cluster_uid_filter']= enumerate_options.get('enumerate_options.cluster_uid_filter')
+        params['enumerate_options.owners']= enumerate_options.get('enumerate_options.owners')
+        params['enumerate_options.backup_object_type']= enumerate_options.get('enumerate_options.backup_object_type')
+        params['enumerate_options.status']= enumerate_options.get('enumerate_options.status')
+        params['enumerate_options.time_range.start_time']= enumerate_options.get('enumerate_options.time_range.start_time')
+        params['enumerate_options.time_range.end_time']= enumerate_options.get('enumerate_options.time_range.end_time')
+    try:
+        response = client.make_request('GET', f"v1/backupschedule/{module.params['org_id']}", params=params)
+        # backup_schedules= response.get('backup_schedules')
+        return response['backup_schedules']
+    except Exception as e:
+        module.fail_json(msg=f"Failed to enumerate Backup Schedule: {str(e)}")
+
+def inspect_backup_schedules(module, client):
+    """Get details of a specific backup location"""
+    params = {
+        'uid': module.params.get('uid')
+    }
+    try:
+        response = client.make_request(
+            'GET',
+            f"v1/backupschedule/{module.params['org_id']}/{module.params['name']}",
+            params=params
+        )
+        return response['backup_schedule']
+    except Exception as e:
+        module.fail_json(msg=f"Failed to inspect Backup Schedule: {str(e)}")
+
+def delete_backup_schedules(module, client):
+    """Delete a backup location"""
+    params = {
+        'uid': module.params.get('uid')
+    }
+    try:
+        response = client.make_request(
+            'DELETE',
+            f"v1/backupschedule/{module.params['org_id']}/{module.params['name']}"
+        )
+        return response, True
+    except Exception as e:
+        module.fail_json(msg=f"Failed to delete Backup Schedule: {str(e)}")
+
+def backup_schedule_request_body(module):
+    """Build the Cloud Credential request object"""
+    backup_schedule_request = {
+        "metadata": {
+            "name": module.params['name'],
+            "org_id": module.params['org_id'],
+            "owner": module.params['owner']
+        },
+        "schedule_policy": module.params['schedule_policy'],
+        "reclaim_policy": module.params['reclaim_policy'],
+        "backup_location": module.params['backup_location'],
```

Comment:
this is deprecated, use `"backup_location_ref": module.params.get('backup_location_ref'),`